server/api/src/rsGeneralAdmin.cpp

```diff
@@ -52,6 +53,62 @@ namespace
         return std::find_if(b, e, [](unsigned char _ch) { return ::isspace(_ch); }) != e;
     } // contains_whitespace
 
+    auto throw_if_downgrading_irods_service_account_rodsadmin(RsComm& rsComm,
+                                                              const std::string_view _option,
+                                                              const std::string_view _user_name,
+                                                              const std::string_view _new_user_type) -> void
+    {
+        if ("rodsadmin" == _new_user_type || "type" != _option) {
+            return;
+        }
+
+        BytesBuf* bbuf = nullptr;
+        irods::at_scope_exit free_buf{[&bbuf] { std::free(bbuf); }};
+
+        if (const auto ec = rsZoneReport(&rsComm, &bbuf); ec < 0) {
+            const auto msg =
+                fmt::format("[{}:{}] - Failed to gather rodsadmin users managing a server in the local zone.",
+                            __func__,
+                            __LINE__);
+            addRErrorMsg(&rsComm.rError, ec, msg.c_str());
+            THROW(ec, msg);
+        }
+
+        const auto buf = static_cast<char*>(bbuf->buf);
+        try {
```

Comment:
Swap lines so that the variable is inside the try-block. `buf` is only used by the call to `nlohmann::json::parse`.